### PR TITLE
Adding the Interplolation of the Orientation

### PR DESCRIPTION
**Pull request by philipp1604**
## Description
In this pull request the interpolation of the orientation was added to the interpolation class. If the user adds a start and end orientation to the interpolation command, the orientation will be interpolated in the output tool path. Additionally a interpolation demonstration class was created

## Motivation and Context

 https://github.com/WBK-Robotics/pybullet_industrial/issues/47

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/pybullet_industrial/interpolation.py [hunk 1]
```diff
@@ -118,34 +138,73 @@ def circular_interpolation(start_point: np.array, end_point: np.array,
...
+    """Interpolates between two orientations
+
+    Args:
+        start_orientation (np.array(3,1)): 3-D array containing the start orientation
+        end_orientation (np.array(3,1)): 3-D array containing the end orientation
+        samples (int): The number of samples used to interpolate
+    """
+    start_orientation = p.getEulerFromQuaternion(start_orientation)
+    end_orientation = p.getEulerFromQuaternion(end_orientation)
+    euler_orientations = np.linspace(start_orientation,
```
Comment (liquidcronos) [inline src/pybullet_industrial/interpolation.py:165]: What happens if the Euler angles cross over from -2pi to 0. It should be checked that this behaves as expected

File: src/pybullet_industrial/interpolation.py [hunk 2]
```diff
@@ -118,34 +138,73 @@ def circular_interpolation(start_point: np.array, end_point: np.array,
...
-    path[0] = cs_x(cs_s)
-    path[1] = cs_y(cs_s)
-    path[2] = cs_z(cs_s)
+    positions[0] = cs_x(cs_s)
+    positions[1] = cs_y(cs_s)
+    positions[2] = cs_z(cs_s)
+
+    final_path = ToolPath(positions=positions)
+
+    if start_orientation is not None and end_orientation is not None:
```
Comment (liquidcronos) [inline src/pybullet_industrial/interpolation.py:205]: This is unexpected behavior. If the positions are interpolated as splines so should the orientations.

This requires splines that can deal with the jump from 0 to -2pi!

**Review by liquidcronos — CHANGES_REQUESTED** [review]
Looks good, however the spline interpolation does not quite behave as expected. We should also add a test to check that behavior is as expected when we are jumping between the angle gap

File: src/pybullet_industrial/interpolation.py (hunk 2 above)
Comment (philipp1604) [reply src/pybullet_industrial/interpolation.py]: @liquidcronos @MaHajo in the new push of my code the spline interpolation was reset to the default code. There are issues that have been encountered using this method. For verification you can check the open issue: https://github.com/WBK-Robotics/pybullet_industrial/issues/50

**Pull request merged**